abcclassroom/dirsetup.py

```diff
@@ -0,0 +1,94 @@
+"""
+abc-classroom.dir-setup
+========================
+
+"""
+
+import os
+import shutil
+import argparse
+import abcclassroom
+
+
+def path_to_example(dataset):
+    """
+    Gets the path to an example dataset.
+
+    Parameters
+    ----------
+    dataset : string
+        stringy boi
+    """
+    abcclassroom_path = os.path.split(abcclassroom.__file__)[0]
+    data_dir = os.path.join(abcclassroom_path, "example-data")
+    data_files = os.listdir(data_dir)
+    if dataset not in data_files:
+        raise KeyError(dataset + " not found in abc-classroom example data.")
+    return os.path.join(data_dir, dataset)
+
+
+def directory_setup():
+    """
+    Construct a file path to an example dataset.
+    This file defines helper functions to access data files in this directory,
+    to support examples. Adapted from the PySAL package.
+
+    Parameters
+    ----------
+    dataset: string
+        Name of a dataset to access (e.g., "epsg.json", or "RGB.byte.tif")
+
+    Returns
+    -------
+        A file path (string) to the dataset
+    """
+    # Creating the default names for the cloned and template repos.
+    cloned = "cloned_repos"
+    template = "template_repos"
+    course = "course_dir"
+    # Making sure the configuration file is where it's supposed to be.
+    config = path_to_example("sample_config.yml")
+    if not os.path.exists(config):
+        raise ValueError(
+            "Configuration file can't be located, please ensure abc-classroom has been installed correctly"
+        )
+    # Allows users to rename the cloned and template repos.
+    parser = argparse.ArgumentParser(description=directory_setup.__doc__)
+    parser.add_argument(
+        "--course_repo",
```

Comment:
We should be consistent and use the same names here and in the config (is `course_repo` the `course_directory` or `course_name` or `short_coursename` or it is using the supplied string for more than one?)

Reply:
totally. `course_repo` i wonder if nathan intended this to be `course_name`? it would make sense to have a course_name argument given we do want a course name for the parent directory.

Reply:
what if we just have a course_name argument and if we need to an org name? but they could config the org name later too... i'm open. 
@nkorinek @kcranston

Reply:
This was intended to be `course_directory`. It was the arguement that would be taken to modify the main directory's name.

Reply:
cool. so what if the arguments are
1. course_name and
2. org_name 

? 
it can then populate the config with those two items, create a dir with the course name as the dir name and then the other 2 dirs are standard names (not customizable via cli)

thoughts @kcranston @nkorinek ?

Reply:
i'm thinking we just use the course-name-here now because the org name is not needed for the setup and there are MANY different things to update in the config file.

Reply:
should the course_name argument be positional (i.e. mandatory) rather than optional (using a flag)? i.e. you would run it as `abc-quickstart coursename` rather than abc-quickstart --course_name coursename`. That's the argparse convention, see https://docs.python.org/3/library/argparse.html#the-add-argument-method .

Reply:
oooh positional i think is the way to go just like assignment-name in the template function. i prefer that.